template/functions/fn_unconscious.sqf

```diff
@@ -0,0 +1,42 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Tyrone
+ * Switches ACRE spoken language on unconsciousness
+ * Call from initPlayerLocal.sqf.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player] call FUNC(unconscious);
+ */
+
+params ["_player"];
+
+if (!hasInterface) exitWith {};
+
+// Set languages
+["en", "English"] call acre_api_fnc_babelAddLanguageType;
+["un", "Unconscious"] call acre_api_fnc_babelAddLanguageType;
+
+// Set spoken
+["en"] call acre_api_fnc_babelSetSpokenLanguages;
+
+// Switch language on unconscious toggle
+["ace_unconscious", {
+    params ["_unit", "_state"];
+    if (_unit != ACE_player) exitWith {}; // Required because of global event
+    if (_state) then {
+        ["un"] call acre_api_fnc_babelSetSpokenLanguages;
+    } else {
+        ["en"] call acre_api_fnc_babelSetSpokenLanguages;
+    };
+}] call CBA_fnc_addEventHandler;
+
+// Handle dying while unconscious
+_player addEventHandler ["Respawn", {
+    ["en"] call acre_api_fnc_babelSetSpokenLanguages;
+}];
```

Comment:
```suggestion
// Switch language on unconscious toggle
["ace_unconscious", {
    params ["_unit", "_state"];

    if (_unit != ACE_player) exitWith {}; // Required because of global event

    if (_state) then {
        ["un"] call acre_api_fnc_babelSetSpokenLanguages;
    } else {
        ["en"] call acre_api_fnc_babelSetSpokenLanguages;
    };
}] call CBA_fnc_addEventHandler;

// Handle unit change (including death)
["unit", {
    params ["_newUnit", "_oldUnit"];

    if (call CBA_fnc_getActiveFeatureCamera != "") exitWith {}; // Ignore if feature camera active (eg. Zeus)

    if (_newUnit getVariable ["ACE_isUnconscious", false]) then {
        ["un"] call acre_api_fnc_babelSetSpokenLanguages;
    } else {
        ["en"] call acre_api_fnc_babelSetSpokenLanguages;
    };
}, false] call CBA_fnc_addPlayerEventHandler;

// Handle feature camera (eg. Zeus)
["featureCamera", {
    params ["_unit", "_newCamera"]; 
    
    if (_newCamera == "" && {ACE_player getVariable ["ACE_isUnconscious", false]})
        ["un"] call acre_api_fnc_babelSetSpokenLanguages; 
    } else { 
        ["en"] call acre_api_fnc_babelSetSpokenLanguages; 
    }; 
}, false] call CBA_fnc_addPlayerEventHandler;
```
Should handle everything, needs testing.

Reply:
Testing today.

Reply:
> 12:53:26 Error in expression <Variable ["ACE_isUnconscious", false]})
> ["un"] call acre_api_fnc_babelSetSpokenL>
> 12:53:26   Error position: <["un"] call acre_api_fnc_babelSetSpokenL>
> 12:53:26   Error Missing ;

Reply:
Line 58, try fixing it.